### PR TITLE
Upgrade to gradle 7.0.

### PR DESCRIPTION
**Pull request by tokuhirom**
- Switch to `maven-publish` plugin from `maven` plugin. In gradle 7, Gradle deprecates the `maven` plugin.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -243,59 +243,58 @@ task codeCoverageReport(type: JacocoReport) {
...
 
         signing {
             required { isReleaseBuild }
             sign configurations.archives
         }
 
-        uploadArchives {
+        publishing {
             repositories {
-                if (isDevBuild) {
```
Comment (tokuhirom) [inline build.gradle:259]: in `maven-publish` plugin, there's a `publishToMavenLocal` task.

File: build.gradle [hunk 2]
```diff
@@ -243,59 +243,58 @@ task codeCoverageReport(type: JacocoReport) {
     }
 }
 
+// https://docs.gradle.org/6.7.1/userguide/publishing_maven.html
```
Comment (imasahiro) [inline build.gradle:246]: What does this comment help for?

**Review by imasahiro — COMMENTED** [review]
LGTM except one comment

File: build.gradle (hunk 2 above)
Comment (tokuhirom) [reply build.gradle:246]: ah it's just a note for writing this PR. I'll remove it.

**Review by imasahiro — APPROVED** [review]

**Pull request merged**